packages/autocomplete-preset-algolia/src/highlight/parseAlgoliaHitReverseHighlight.ts

```diff
@@ -1,12 +1,11 @@
-import { Hit } from '@algolia/client-search';
-
+import { HighlightedHit } from './HighlightedHit';
 import { parseAlgoliaHitHighlight } from './parseAlgoliaHitHighlight';
 import { ParseAlgoliaHitParams } from './ParseAlgoliaHitParams';
 import { ParsedAttribute } from './ParsedAttribute';
 import { reverseHighlightedParts } from './reverseHighlightedParts';
 
-export function parseAlgoliaHitReverseHighlight<THit extends Hit<{}>>(
-  props: ParseAlgoliaHitParams<THit>
-): ParsedAttribute[] {
+export function parseAlgoliaHitReverseHighlight<
+  THit extends HighlightedHit<unknown>
```

Comment:
You suggested Record<string, unknown> in message, any reason why you changed it to unknown again?

Reply:
Given:

```ts
import { HighlightResult } from '@algolia/client-search';

export type HighlightedHit<THit> = THit & {
  _highlightResult?: HighlightResult<THit>;
};
```

I tried `HighlightResult<Record<string, unknown>>` with a DocSearch hit and it doesn't work because properties can be `null` (e.g., `content`).

We need to improve the client's [`HighlightResult`](https://github.com/algolia/algoliasearch-client-javascript/blob/4591d0eb02382b4380cb0a0a124461315d5b1bc2/packages/client-search/src/types/Hit.ts#L8-L12) type to allow this.